lib/absinthe/schema/notation/sdl_render.ex

```diff
@@ -0,0 +1,361 @@
+    block(
+      "schema",
+      Enum.map(schema.field_definitions, &render/1)
+    )
+  end
+
+  # TODO: remove once macro schema has SchemaDeclaration
+  def render(%{
+        query: query_type,
+        mutation: mutation_type,
+        subscription: subscription_type
+      }) do
+    schema_type_docs =
+      [
+        query_type && concat("query: ", string(query_type.name)),
+        mutation_type && concat("mutation: ", string(mutation_type.name)),
+        subscription_type && concat("subscription: ", string(subscription_type.name))
+      ]
+      |> Enum.reject(&is_nil/1)
+
+    block(
+      "schema",
+      schema_type_docs
+    )
+  end
+
+  def render(%Schema.InputValueDefinition{} = input_value) do
+    concat([
+      string(input_value.name),
+      ": ",
+      render(input_value.type),
+      default(input_value.default_value)
+    ])
+    |> description(input_value.description)
+  end
+
+  def render(%Schema.FieldDefinition{} = field) do
+    # adapter = blueprint.adapter || Absinthe.Adapter.LanguageConventions
+    adapter = Absinthe.Adapter.LanguageConventions
+
+    concat([
+      string(adapter.to_external_name(field.name, :field)),
+      arguments(field.arguments),
+      ": ",
+      render(field.type)
+    ])
+    |> deprecated(field.deprecation)
+    |> description(field.description)
+  end
+
+  # Don't render introspection types
+  # def render(%Schema.ObjectTypeDefinition{name: "__" <> _introspection_type}) do
+  #   empty()
+  # end
+
+  # Don't render builtin scalar types
+  # def render(%Schema.ObjectTypeDefinition{identifier: identifier})
+  #     when identifier in @builtin_scalars do
+  #   empty()
+  # end
+
+  def render(%Schema.ObjectTypeDefinition{} = object_type) do
+    block(
+      "type",
+      concat([
+        string(object_type.name),
+        implements(object_type.interface_types)
+      ]),
+      Enum.map(object_type.fields, &render/1)
+    )
+    |> description(object_type.description)
+  end
+
+  def render(%Schema.InputObjectTypeDefinition{} = input_object_type) do
+    block(
+      "input",
+      string(input_object_type.name),
+      Enum.map(input_object_type.fields, &render/1)
+    )
+    |> description(input_object_type.description)
+  end
+
+  def render(%Schema.UnionTypeDefinition{} = union_type) do
+    possible_type_docs = Enum.map(union_type.types, & &1.name)
+
+    concat([
+      "union ",
+      string(union_type.name),
+      " = ",
+      join(possible_type_docs, " | ")
+    ])
+    |> description(union_type.description)
+  end
+
+  def render(%Blueprint.Schema.InterfaceTypeDefinition{} = interace_type) do
+    block(
+      "interface",
+      string(interace_type.name),
+      Enum.map(interace_type.fields, &render/1)
+    )
+    |> description(interace_type.description)
+  end
+
+  def render(%Blueprint.Schema.EnumTypeDefinition{} = enum_type) do
+    block(
+      "enum",
+      string(enum_type.name),
+      Enum.map(enum_type.values, &string(&1.name))
```

Comment:
* [ ] Enum *values* can be deprecated too

Reply:
@bruce I don't think SDL is setup correctly to allow attaching the `deprecated` directive to enum values.

I added this to the other directive location PR: https://github.com/absinthe-graphql/absinthe/pull/743

Reply:
This should be fixed now that #741 is merged.